deeplearning4j-nn/src/main/java/org/deeplearning4j/nn/graph/ComputationGraph.java

```diff
@@ -3183,6 +3185,20 @@ protected void clearLayersStates() {
     }
 
     /**
+     * Increment the epoch count (in the underlying {@link MultiLayerConfiguration} by 1).
+     * Note that this is done <i>automatically</i> when using iterator-based fitting methods, such as
+     * {@link #fit(DataSetIterator)} or {@link #fit(MultiDataSet)}. However, when using non-iterator fit methods
+     * (DataSet, MultiDataSet, INDArrays etc), the network has no way to know when one epoch ends and another starts.
+     * In such situations, this method can be used to increment the epoch counter.<br>
+     * Note that the epoch counter is used for situations such as some learning rate schedules, and the like.
+     *
+     * The current epoch count can be obtained using {@code ComputationGraph.getConfiguration().getEpochCount()}
+     */
+    public void incrementEpochCount(){
```

Comment:
Is this thread-safe, could we envision a situation with concurrent `setEpochCount` calls ?

Reply:
No, it's not thread safe - but then nothing in MultiLayerNetwork or ComputationGraph is.
Setting can be done directly on the configuration objects.